src/struct_ls/cyclic_reduction.c

```diff
@@ -501,7 +501,7 @@ hypre_CyclicReductionSetup( void               *cyc_red_vdata,
    HYPRE_Int               flop_divisor;
 
    HYPRE_Int               x_num_ghost[] = {0, 0, 0, 0, 0, 0};
-#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP)
+#if 0 //defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP)
```

Comment:
Okay, so now I am confused again.  Why is this in master but being commented out here?

Reply:
@rfalgout @ulrikeyang This, and the other "#if 0" in the PR, were intended to completely disable an (incomplete) feature of having "CPU-GPU" switching levels in SMG and PFMG (the `data_location` is for that). We introduced it long time ago but it seems not always working (have not investigated to figure out why). We commented it out its usage in the driver and not suggested it. But some smart user figured out its meaning, tried to use it, and had seg-fault issue. I just want to #ifdef it out for now. If you don't like it, we change it back. Thanks!

Reply:
Okay, I understand.  Basically, this is a feature in waiting that will eventually either be added back in or abandoned.  This is fine.  Thanks!